src/landscape/C4Particles.h

```diff
@@ -157,12 +157,12 @@ class C4ParticleValueProvider
 	bool IsConstant() const { return isConstant; }
 	bool IsRandom() const { return valueFunction == &C4ParticleValueProvider::Random; }
 	C4ParticleValueProvider() :
-		startValue(0.f), endValue(0.f), currentValue(0.f), rerollInterval(0), smoothing(0), keyFrameCount(0), valueFunction(0), isConstant(true), floatValueToChange(0), typeOfValueToChange(VAL_TYPE_FLOAT)
+		startValue(0.f), endValue(0.f), currentValue(0.f), rerollInterval(0), smoothing(0), keyFrameCount(0), valueFunction(nullptr), isConstant(true), floatValueToChange(nullptr), typeOfValueToChange(VAL_TYPE_FLOAT)
```

Comment:
My copy of clang-tidy (on Windows) crashes consistently for some files in the project:
- src/landscape/C4Particles.cpp
- src/script/C4AulDebug.cpp

The Linux one succeeded though; I'm not sure why it didn't modify these. I'll run that again too.